server/app/driver/driver.py

```diff
@@ -65,10 +65,10 @@ def filter_dataframe(self, filter):
         pass
 
     @abstractmethod
-    def annotation(self, df, axis, fields=None):
+    def annotation(self, filter, axis, fields=None):
         """
         Gets annotation value for each observation
-        :param df: from filter_cells, dataframe
+        :param filter: filter: dictionary with filter params
```

Comment:
probably should be:
`:param filter: dictionary with filter params`

also, maybe consider rewording to: dictionary with XXX as keys and YYY as the values.

Reply:
hmm, this is a little more complicated. I think I'm going to punt on this and add a ticket to make a filter object to simplify documentation in the long term.